retail/retail-clickstream-application/README.md

```diff
@@ -0,0 +1,234 @@
+
+![Cloud Run Proxy](cloud-run-proxy.png)
+
+This repo contains example code to show how to send data from a ecommerce website to Pub/Sub using Google Tag Manager, Cloud Run and Pub/Sub.
+
+The data structure uses the [Data Layer Ecommerce](https://developers.google.com/tag-manager/ecommerce-ga4) format recommended for Google Tag Manager
+
+## Git clone repo
+
+```
+git clone https://github.com/GoogleCloudPlatform/dataflow-sample-applications.git
+cd dataflow-sample-applications/retail/retail-clickstream-application
+```
+
+## Set-up Cloud Environment
+
+### Initilize your account and project
+
+```shell
+gcloud init
+```
+
+### Set Google Cloud Project
+
+```
+export GOOGLE_CLOUD_PROJECT=my-project-id
+gcloud config set project $GOOGLE_CLOUD_PROJECT
+```
+
+### Enable Google Cloud APIs
+
+```
+gcloud services enable compute.googleapis.com cloudbuild.googleapis.com artifactregistry.googleapis.com
+```
+
+### Set compute zone
+
+```
+gcloud config set compute/zone us-central1-f
+```
+
+# Build container
+
+```
+export RUN_SOURCE_DIR=cloud-run-pubsub-proxy
+
+gcloud builds submit $RUN_SOURCE_DIR --tag gcr.io/$GOOGLE_CLOUD_PROJECT/pubsub-proxy
+```
+
+### List containers
+
+Check that the container was succesfully created
+
+```
+gcloud container images list
+```
+
+You should see the following output:
+
+```
+NAME
+gcr.io/my-project-id/pubsub-proxy
+Only listing images in gcr.io/my-project-id. Use --repository to list images in other repositories.
+```
+
+
+## Deploy using Terraform
+
+Use Terraform to deploy the folllowing services defined in the `main.tf` file
+
+- Cloud Run
+- Pub/Sub
+- Google Cloud Storage
+- Dataflow
+- BigQuery
+
+### Install Terraform
+
+Follow the instructions to [install the Terraform cli](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/gcp-get-started)
+
+### Update Project ID in terraform.tfvars
+
+Rename terraform.tfvars.example file to terraform.tfvars and update the default project ID in terraform.tfvars file to match your project ID
+
+```
+sed "-i" "" "-e" 's/default-project-id/'"$GOOGLE_CLOUD_PROJECT"'/g' terraform.tfvars
+```
+
+### Initialize Terraform
+
+```
+terraform init
+```
+
+You should see the following output
+
+```
+Initializing the backend...
+
+Initializing provider plugins...
+- Checking for available provider plugins...
+- Downloading plugin for provider "google" (hashicorp/google) 3.48.0...
+
+Terraform has been successfully initialized!
+
+You may now begin working with Terraform. Try running "terraform plan" to see
+any changes that are required for your infrastructure. All Terraform commands
+should now work.
+
+If you ever set or change modules or backend configuration for Terraform,
+rerun this command to reinitialize your working directory. If you forget, other
+commands will detect it and remind you to do so if necessary.
+```
+
+### Create resoureces in Google Cloud
+
+*Note: Since Terraform 0.12 and above you no longer need to run `terraform plan` first.*
```

Comment:
BTW I think as this is your version 1, its ok to set a base version that must be used.

Reply:
I have left a note to state what version this was tested on.